### PR TITLE
#165052923 An admin can add a department 

### PR DESCRIPTION
**Pull request by Thuku**
## What does this PR do?
 - creates departments reducer
 - creates departments actions
 - adds departments reducer to root reducer
 - creates departments component
 - creates departments container
 - creates add department modal component
 - creates add department modal container
## Description of Task to be completed?

``` Given an admin with access
When I Visit the home page
Then I should see a card `Departments` next to Andela centres. 
When I click on the card
Then it should direct me to a list of departments
that has a button to allow me to add a department.
```

## How should this be manually tested?

 - on the home page click departments card

## What are the relevant pivotal tracker stories?
- [#165052923](https://www.pivotaltracker.com/story/show/165052923)

## Packages installed
 - none

## Deployment note
 - none
## Screenshots
 - departments card
Screenshot 2019-04-04 at 23 47 21

- departments list

Screenshot 2019-04-04 at 23 47 33

- add departments modal

Screenshot 2019-04-04 at 23 47 43

### CODE REVIEW

**Comment by Thuku** [comment]
@andela-fojuri @EmaEvidence Kindly review.

File: src/__test__/reducers/departments.reducer.test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  it('should handle CREATE_DEPARTMENT_FAILURE', () => {
+    expect(departmentsReducer(state, {
+      type: CREATE_DEPARTMENT_FAILURE,
+      payload: 'Could not update the centre.'
+    }))
+      .toEqual(expect.objectContaining({
+        isLoading: false,
+        createFailure: 'Could not update the centre.',
```
Comment (fojuri) [inline src/__test__/reducers/departments.reducer.test.js]: `Could not update the centre` does not depict what is being tested

File: src/__test__/components/AndelaDepartmentsModal.test.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  it('handles form submission', () => {
+    const wrapper = shallow(<DepartmentsModal {...props} />);
+    const handleSubmitSpy = jest.spyOn(wrapper.instance(), 'handleSubmit');
+
+    wrapper.instance().handleSubmit();
+    expect(handleSubmitSpy).toHaveBeenCalled();
+  });
+
+
```
Comment (fojuri) [inline src/__test__/components/AndelaDepartmentsModal.test.js]: Remove one line

File: src/components/Departments/DepartmentsComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+              reset={resetMessage}
+            />
+          )}
+
+          {isLoading && !showStatus && <LoaderComponent />}
+
+          {!isLoading && hasDepartments && (
+            <Cards
+              data={departmentsList}
+              headings={['name', 'id']}
```
Comment (fojuri) [inline src/components/Departments/DepartmentsComponent.jsx]: Why do we have the `id` on the heading?

Comment (Thuku) [reply src/components/Departments/DepartmentsComponent.jsx]: it is the number (id) displayed on the card below the department name

Comment (fojuri) [reply src/components/Departments/DepartmentsComponent.jsx]: Yes, I was asking if there's a reason to display the id below the name.

**Comment by Thuku** [comment]
> * Please close the modal after successful creation of a department
> * Add more tests to increase coverage
> * Update  the branch.

@andela-fojuri There was a discussion on closing the modals and we went with keeping them open in the case a user is adding a lot of items.

**Pull request merged**